### PR TITLE
[Renaming] Skip RenameMethodRector when target method is __invoke with private/protected modifier

### PR DESCRIPTION
**Pull request by samsonasik**
Closes #5511

### CODE REVIEW

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba it is ready for review.

File: rules/renaming/src/Rector/MethodCall/RenameMethodRector.php [hunk 1]
```diff
@@ -166,6 +173,15 @@ private function shouldSkipForExactClassMethodForClassMethod(ClassMethod $classM
             return false;
         }
 
-        return $classMethod->getAttribute(AttributeKey::CLASS_NAME) === $type;
+        $isExactClassMethodForClasssMethod = $classMethod->getAttribute(AttributeKey::CLASS_NAME) === $type;
+        if ($isExactClassMethodForClasssMethod) {
+            return true;
+        }
+
+        if ($newMethodName !== '__invoke') {
```
Comment (TomasVotruba) [inline rules/renaming/src/Rector/MethodCall/RenameMethodRector.php:181]: All the magic methods should be handled here, as I comented in the issue:
https://github.com/rectorphp/rector/pull/5511#issuecomment-778503158

Comment (samsonasik) [reply rules/renaming/src/Rector/MethodCall/RenameMethodRector.php]: implemented.

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba I think it is ready.

**Comment by TomasVotruba** [comment]
Looks good, thank you :+1:

**Comment by ruudk** [comment]
Great, thanks @samsonasik

**Pull request merged**